src/config_defaults.ts

```diff
@@ -0,0 +1,171 @@
+function o(object) {
+    return Object.assign(Object.create(null), object)
+}
+
+const DEFAULTS = o({
```

Comment:
Sorry if this is a stupid question but why is this duplicated in `src/common/state.ts`? Would there be a way to have `defaultState` in `state.ts` be derived from `DEFAULTS` in `config_defaults.ts`?
Feel free to ignore this question if it's because the PR is still a WIP :).